### PR TITLE
[react-datepicker] upgrade react datepicker v4 & popper v2

### PR DESCRIPTION
**Pull request by JanStevens**
Based on: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53588 & https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53519 but upgraded to popper v2 as requested

Note: I upped the minimal node version

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/Hacker0x01/react-datepicker/blob/master/src/index.jsx#L141
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/react-datepicker/tslint.json [hunk 1]
```diff
@@ -1 +1,6 @@
-{ "extends": "dtslint/dt.json" }
+{
+  "extends": "dtslint/dt.json",
+  "rules": {
+    "no-redundant-undefined": false
```
Comment (elibarzilay) [inline types/react-datepicker/tslint.json:4]: why is this needed?

File: types/react-datepicker/index.d.ts [hunk 2]
```diff
@@ -15,10 +15,11 @@
 //                 Walter Kennedy <https://github.com/wthefourth>
 //                 Shiftr Tech SAS <https://github.com/ShiftrTechSAS>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+// Minimum TypeScript Version: 3.8
+// TypeScript Version: 3.8
```
Comment (elibarzilay) [inline types/react-datepicker/index.d.ts:19]: There's no need for both

**Comment by Newbie012** [comment]
can this be merged?
